### PR TITLE
add parse(CharSequence) support for the implicit converter

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang@uk.ibm.com>

### CODE REVIEW

File: tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java [hunk 1]
```diff
@@ -78,6 +79,14 @@ public void testImplicitConverterStringValueOf() {
         Assert.assertNotNull(value);
         Assert.assertEquals(value.getVal(), "stringValueOf");
     }
+    
+    @Test
+    public void testImplicitConverterCharSequenceParse() {
+        ConvTestTypeWCharSequenceParse value = config.getValue("tck.config.test.javaconfig.converter.implicit.charSequenceParse",
```
Comment (jmesnil) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java]: let's use an actual class (e.g. java.time.Year). We don't need to create a weirdly named class to check that implicit converter for Java runtime classes work

Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java]: added more tests to cover your usage.

File: api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java [hunk 2]
```diff
@@ -97,6 +97,7 @@
  * <ul>
  *     <li>The target type {@code T} has a Constructor with a String parameter, or</li>
  *     <li>the target type {@code T} has a {@code static T valueOf(String)} method, or</li>
+ *     <li>the target type {@code T} has a {@code static T parse(CharSequence)} method, or</li>
```
Comment (johnament) [inline api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java]: Unneeded or

File: tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java [hunk 3]
```diff
@@ -21,9 +21,13 @@
 
 import static org.eclipse.microprofile.config.tck.base.AbstractTest.addFile;
 
+import java.time.YearMonth;
```
Comment (johnament) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java]: This test is also broken, ProcessAnnotatedType is not fired for fields in an Arquillian test.

Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java]: Which bit is broken?

File: api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java (hunk 2 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java]: fixed

File: tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java (hunk 3 above)
Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ImplicitConverterTest.java]: used the classes to hold the field injection instead. This must have something to do with the Arquillian integrator you are using.

**Pull request merged**